BLUESPAWN-win-client/src/scan/FileScanner.cpp

```diff
@@ -0,0 +1,256 @@
+    std::vector<std::wstring> strings{};
+
+    DWORD dwStringStart{};
+    for(DWORD idx = 0; idx < data.GetSize(); idx++) {
+        if(!(data[idx] >= 0x20 && data[idx] <= 0x7F)) {
+            DWORD dwStringLength = idx - dwStringStart;
+            if(dwStringLength >= dwMinLength) {
+                strings.emplace_back(
+                    StringToWidestring(std::string{ PCHAR(LPVOID(data)) + dwStringStart, dwStringLength }));
+            }
+
+            dwStringStart = idx + 1;
+        }
+    }
+
+    auto dwStringLength = data.GetSize() - dwStringStart;
+    if(dwStringLength >= dwMinLength) {
+        strings.emplace_back(StringToWidestring(std::string{ PCHAR(LPVOID(data)) + dwStringStart, dwStringLength }));
+    }
+
+    dwStringStart = 0;
+    auto mem{ reinterpret_cast<PWCHAR>(LPVOID(data)) };
+    for(DWORD idx = 0; 2 * idx < data.GetSize(); idx++) {
+        if(!(mem[idx] >= 0x20 && mem[idx] < 0x7E)) {
+            dwStringLength = idx - dwStringStart;
+            if(dwStringLength >= dwMinLength) {
+                strings.emplace_back(std::wstring{ PWCHAR(LPVOID(data)) + dwStringStart, dwStringLength });
+            }
+
+            dwStringStart = idx + 1;
+        }
+    }
+
+    dwStringLength = data.GetSize() / 2 - dwStringStart;
+    if(dwStringLength >= dwMinLength && data.GetSize() / 2 > dwStringStart) {
+        strings.emplace_back(std::wstring{ mem + dwStringStart, dwStringLength });
+    }
+
+    return strings;
+}
+
+std::vector<std::wstring> FileScanner::ExtractFilePaths(IN CONST std::vector<std::wstring>& strings) {
+    std::vector<std::wstring> filepaths{};
+    std::wregex regex{ L"[a-zA-Z]:([/\\\\][a-zA-Z0-9(). @_-]+)+" };
+    for(auto& string : strings) {
+        std::wsmatch match{};
+        if(std::regex_search(string, match, regex)) {
+            for(auto& filename : match) {
+                if(FileSystem::CheckFileExists(filename.str())) {
+                    filepaths.emplace_back(filename.str());
+                }
+            }
+        }
+    }
+    return filepaths;
+}
+
+void FileScanner::UpdateModules() {
+    BeginCriticalSection _{ hGuard };
+
+    FILETIME time{};
+    GetSystemTimeAsFileTime(&time);
+
+    uint64_t tdiff{ (static_cast<uint64_t>(time.dwHighDateTime - ModuleLastUpdateTime.dwHighDateTime) << 32) +
+                    time.dwLowDateTime - ModuleLastUpdateTime.dwLowDateTime };
+    DWORD dwSecondsElapsed = tdiff / 10000;
+    if(dwSecondsElapsed >= MODULE_UPDATE_INTERVAL) {
+        modules.clear();
+        hashes.clear();
+
+        std::vector<DWORD> processes(1024);
+        DWORD dwBytesNeeded{};
+        auto success{ EnumProcesses(processes.data(), 1024 * sizeof(DWORD), &dwBytesNeeded) };
+        if(dwBytesNeeded > 1024 * sizeof(DWORD)) {
+            processes.resize(dwBytesNeeded / sizeof(DWORD));
+            success = EnumProcesses(processes.data(), dwBytesNeeded, &dwBytesNeeded);
```

Comment:
Codacy found an issue: [Variable 'success' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5982821)